### ISSUE TITLE
Not possible to supply a password when using the high level interfaces

### ISSUE THREAD

**Issue by ptallett** [adamhathcock/sharpcompress#37]
When processing a password protected ZIP file using the high level IArchiveReader/Writer interfaces, it is not possible to supply the password as this functionality is only exposed by lower level interfaces.

Cheers,
Paul

**Comment by sepehr1014** [comment]
Any plans on adding this functionality?

**Comment by ptallett** [comment]
Its fairly easy to hack this. I just put a static string in the lower level class and a set method in the high level API. Not something I’m proud of though, so I didn’t check it in! :)

 

Cheers,

Paul

**Comment by sepehr1014** [comment]
Thanks @ptallett but in our situation, we don't know if the files are password protected beforehand. There needs to be an event which is raised on password protected archives to get the password and continue extracting.

**Comment by ptallett** [comment]
Yes, that would be ideal, but you can workaround it by trying to open the ZIP file without a password, catching the exception, requesting a password, then retrying to open the ZIP file. Works a treat. You need to do something similar if the password is incorrect in any case.

 

The drawback with this approach is each file in a ZIP container can have a different password, but I just ignored that scenario as its rare.

 

Cheers,

Paul
